geomesa-geojson/geomesa-geojson-api/src/main/scala/org/locationtech/geomesa/geojson/query/GeoJsonQuery.scala

```diff
@@ -99,55 +120,64 @@ object GeoJsonQuery {
           Or(ors: _*)
 
         case (prop, v: JValue) =>
-          Equals(JsonPathParser.parse(prop), v.values)
+          Equals(checkJsonPaths(prop), v.values)
+      }
+      if (predicates.length > 1) {
+        And(predicates: _*)
+      } else {
+        predicates.head
       }
-      if (predicates.length > 1) { And(predicates: _*) } else { predicates.head }
     }
   }
 
   /**
     * Evaluate a complex predicate
     *
-    * @param path path that the predicate operates on
+    * @param prop property that the predicate operates on
     * @param json predicate json object
     * @return
     */
-  private def evaluatePredicate(path: Seq[PathElement], json: JObject): GeoJsonQuery = {
+  private def evaluatePredicate(prop: String, json: JObject): GeoJsonQuery = {
     json.obj.headOption match {
       case Some(("$bbox", v: JArray)) =>
         // { "$bbox" : [-180, -90, 180, 90] }
         val List(xmin, ymin, xmax, ymax) = v.values.asInstanceOf[List[Number]]
-        Bbox(path, xmin.doubleValue, ymin.doubleValue, xmax.doubleValue, ymax.doubleValue)
+        Bbox(prop, xmin.doubleValue, ymin.doubleValue, xmax.doubleValue, ymax.doubleValue)
 
       case Some(("$intersects", v: JObject)) =>
         // { "$intersects" : { "$geometry" : { "type" : "Point", "coordinates" : [30, 10] } } }
-        Intersects(path, evaluateGeometry(v))
+        Intersects(prop, evaluateGeometry(v))
 
       case Some(("$within", v: JObject)) =>
         // { "$within" : { "$geometry" : { "type" : "Polygon", "coordinates": [ [ [0,0], [3,6], [6,1], [0,0] ] ] } } }
-        Within(path, evaluateGeometry(v))
+        Within(prop, evaluateGeometry(v))
+
+      case Some(("$dwithin", v: JObject)) =>
+        // { "$dwithin" : { "$geometry" : { "type" : "Point", "coordinates" : [30, 10] }, "$dist" : 100.50, "$unit" : "feet" } }
```

Comment:
My main goal is to point out that (unfortunately) there are dragons here.  Since part of the GeoTools API drops the units and uses the CRS's units, one can pick up degrees.  Since GeoMesa likes meters, some query pathways may re-write dwithin assuming meters using the GeodeticCalculator. 

Effectively, dwithin hits the difference between geometry and geography.